lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -302,6 +302,33 @@ static bool recursiveType(const StructType *ST, const Type *Ty) {
   return Run(Ty);
 }
 
+// Add decoration if needed
```

Comment:
Good catch. I do not know why I was returning the same pointer that I pass in.
I suppose there are no other changes here?

Thanks

Reply:
Thanks for the review!

It wasn't wrong or anything it just turns out we didn't actually need it.

The only other changes to the function are:
1) Instead of asserting that an `IntrinsicInst` is an `fpbuiltin,` just skip them if they are not so that we can call this function in all cases

2) Handle fpmath metadata